### PR TITLE
ENH: Create ChatGPTHandler. Fixes #734

### LINKED ISSUE #734: Create a WebsocketHandler subclass to interact with OpenAI ChatGPT chat completion API

### PR DESCRIPTION
**Pull request by sanand0**

### CODE REVIEW

**Review by jaidevd — APPROVED** [review]
Just curious as to why this can't be done with a ProxyHandler.

But otherwise, LGTM.

**Comment by sanand0** [comment]
There are 2 reasons why we created a handler for this.

1. **Converting server side events to websocket messages**. OpenAI allows streaming word by word, but that uses server-side events. It's hard to parse each token from a HTTP stream in JavaScript and render it word by word. Websockets make it easier. ChatGPTHandler acts as a converter.
2. **Message history**. ChatGPTHandler stores the previous messages and automatically sends it to OpenAI. It simplifies an otherwise-common step.

That apart, yes, ProxyHandler is a good alternative.

**Comment by sanand0** [comment]
Closed in favor of #742 which adds test cases and a few minor feature
